sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3014,7 +3014,10 @@ class SQLConf extends Serializable with Logging {
 
   def csvColumnPruning: Boolean = getConf(SQLConf.CSV_PARSER_COLUMN_PRUNING)
 
-  def legacySizeOfNull: Boolean = getConf(SQLConf.LEGACY_SIZE_OF_NULL)
+  def legacySizeOfNull: Boolean = {
+    // size(null) should return null under ansi mode.
+    getConf(SQLConf.LEGACY_SIZE_OF_NULL) && !getConf(ANSI_ENABLED)
```

Comment:
Thanks, `ansi` conf overrides `legacy` conf ? Do we have another `legacy` conf like this? Is it consistent with the other `legacy`? Could you update the migration guide and conf document to be consistent, too?

cc @gatorsmile

Reply:
This is the first one, and I don't think we need a migration guide as the behavior doesn't change by default.

I'll update the config doc.

Reply:
Thank you!